cyclonedx_py/parser/environment.py

```diff
@@ -71,16 +78,24 @@ def __init__(self, use_purl_bom_ref: bool = False) -> None:
                 c.author = i_metadata['Author']
 
             if 'License' in i_metadata and i_metadata['License'] != 'UNKNOWN':
-                c.licenses.add(LicenseChoice(license_expression=i_metadata['License']))
-
-            if 'Classifier' in i_metadata:
-                for classifier in i_metadata['Classifier']:
-                    if str(classifier).startswith('License :: OSI Approved :: '):
-                        c.licenses.add(
-                            LicenseChoice(
-                                license_expression=str(classifier).replace('License :: OSI Approved :: ', '').strip()
+                if licence_output_format == 'expression':
+                    c.licenses.add(LicenseChoice(license_expression=i_metadata['License']))
+                else:
+                    c.licenses.add(LicenseChoice(license_=License(license_name=i_metadata['License'])))
+
+            elif 'Classifier' in i_metadata:
```

Comment:
Important change to note: the license identifier in the classifier header will be a fallback in case the license is not available as a mean header. 

Note: This functionality was actually not working. It was looping over each letter of the classifier value instead of going over the different classifier headers.